blockchain/state_transition.go

```diff
@@ -101,7 +102,7 @@ type Message interface {
 
 	// IntrinsicGas returns `intrinsic gas` based on the tx type.
 	// This value is used to differentiate tx fee based on the tx type.
-	IntrinsicGas(currentBlockNumber uint64) (uint64, error)
+	IntrinsicGas(currentBlockNumber uint64, r params.Rules) (uint64, error)
```

Comment:
@yoomee1313 Do you have any reason to pass `params.Rules` instead of `bool` like ethereum?

Reply:
@jeongkyun-oh First of all, i forgot to mention that this PR is part of implementation of tx data fee unification hard fork item. IntrinsicGas will use `r.IsIstanbul` information.
I pass `params.Rules` instead of `bool` for later possible hard fork item. If another boolean needs to be passed for future hard fork, it have to fix the code for every transaction type like this PR does.

Reply:
Thanks for explanation. :)